### PR TITLE
Modify some layouts and add password to user model

### PR DESCRIPTION
**Pull request by khanhkhanh**

### CODE REVIEW

File: app/views/static_pages/home.html.erb [hunk 1]
```diff
@@ -1,4 +1,12 @@
...
+
+  <h2>
+    This is the home page for the
+    <a href="http://www.railstutorial.org/">Ruby on Rails Tutorial</a>
+    sample application.
+  </h2>
+</div>
+
+<%= link_to image_tag("rails.png", alt: "Rails logo"),
+            'http://rubyonrails.org/' %>
```
Comment (leduytungframgia) [inline app/views/static_pages/home.html.erb]: add end line

File: app/views/users/new.html.erb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      <%= f.label :password %>
+      <%= f.password_field :password %>
+
+      <%= f.label :password_confirmation, "Confirmation" %>
+      <%= f.password_field :password_confirmation %>
+
+      <%= f.submit "Create my account", class: "btn btn-primary" %>
+    <% end %>
+  </div>
+</div>
```
Comment (leduytungframgia) [inline app/views/users/new.html.erb]: add end line

File: app/views/users/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+<div class="row">
+  <aside class="col-md-4">
+    <section class="user_info">
+      <h1>
+        <%= gravatar_for @user %>
+        <%= @user.name %>
+      </h1>
+    </section>
+  </aside>
+</div>
```
Comment (leduytungframgia) [inline app/views/users/show.html.erb]: add end line

File: app/views/static_pages/about.html.erb [hunk 4]
```diff
@@ -0,0 +1,2 @@
+<h1>StaticPages#about</h1>
+<p>Find me in app/views/static_pages/about.html.erb</p>
```
Comment (leduytungframgia) [inline app/views/static_pages/about.html.erb]: add end line

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find(params[:id])
+  end
+
+  def new
+  	@user = User.new
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: indent

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  def create
+    @user = User.new(user_params)
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove `()`

File: app/controllers/users_controller.rb [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+    if @user.save
+      flash[:success] = "Welcome to the Education App!"
+      redirect_to @user
+    else
+      render 'new'
+    end
+  end
+
+  private
+
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove

File: app/controllers/users_controller.rb [hunk 8]
```diff
@@ -0,0 +1,26 @@
...
+    def user_params
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: indent

File: app/helpers/users_helper.rb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+module UsersHelper
+
+  # Returns the Gravatar for the given user.
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove all remark

File: app/helpers/users_helper.rb [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+  def gravatar_for(user)
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove `()`

File: app/helpers/users_helper.rb [hunk 11]
```diff
@@ -0,0 +1,9 @@
...
+    gravatar_id = Digest::MD5::hexdigest(user.email.downcase)
+    gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}"
+    image_tag(gravatar_url, alt: user.name, class: "gravatar")
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove `()`

File: app/views/users/new.html.erb [hunk 12]
```diff
@@ -0,0 +1,22 @@
+<% provide(:title, 'Sign up') %>
+<h1>Sign up</h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@user) do |f| %>
+      <%= f.label :name %>
```
Comment (dieunb) [inline app/views/users/new.html.erb]: error message???

File: app/views/users/show.html.erb [hunk 13]
```diff
@@ -0,0 +1,11 @@
+<% provide(:title, @user.name) %>
```
Comment (dieunb) [inline app/views/users/show.html.erb]: remove `()`

File: app/controllers/users_controller.rb [hunk 14]
```diff
@@ -0,0 +1,26 @@
+class UsersController < ApplicationController
+  def show
+    @user = User.find(params[:id])
```
Comment (dieunb) [inline app/controllers/users_controller.rb]: remove `()`

File: app/helpers/users_helper.rb [hunk 15]
```diff
@@ -0,0 +1,8 @@
+module UsersHelper
+  def gravatar_for(user, options = { size: 80 })
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove `()`

File: app/helpers/users_helper.rb [hunk 16]
```diff
@@ -0,0 +1,8 @@
...
+    gravatar_id = Digest::MD5::hexdigest user.email.downcase
+    size = options[:size]
+    gravatar_url = "https://secure.gravatar.com/avatar/#{gravatar_id}?s=#{size}"
+    image_tag(gravatar_url, alt: user.name, class: "gravatar")
```
Comment (dieunb) [inline app/helpers/users_helper.rb]: remove `()`

File: app/views/users/new.html.erb [hunk 17]
```diff
@@ -0,0 +1,24 @@
+<% provide(:title, 'Sign up') %>
+<h1>New account</h1>
+
+<div class="row">
+  <div class="col-md-6 col-md-offset-3">
+    <%= form_for(@user) do |f| %>
```
Comment (dieunb) [inline app/views/users/new.html.erb]: remove `()`

File: app/views/users/new.html.erb [hunk 18]
```diff
@@ -0,0 +1,24 @@
+<% provide(:title, 'Sign up') %>
```
Comment (dieunb) [inline app/views/users/new.html.erb]: remove `()`

File: app/views/shared/_error_messages.html.erb [hunk 19]
```diff
@@ -0,0 +1,12 @@
+<% if @user.errors.any? %>
+  <div id="error_explanation">
+    <div class="alert alert-danger">
+      The form contains <%= pluralize(@user.errors.count, "error") %>.
```
Comment (dieunb) [inline app/views/shared/_error_messages.html.erb]: remove `()`

File: config/routes.rb [hunk 20]
```diff
@@ -1,6 +1,7 @@
 Rails.application.routes.draw do
-  root 'static_pages#home'
-
-  get 'help' => 'static_pages#help'
-
+  root             'static_pages#home'
+  get 'help'    => 'static_pages#help'
+  get 'new_user'  => 'users#new'
```
Comment (dieunb) [inline config/routes.rb]: new_user ??? Why don't `sign_up`

File: config/routes.rb [hunk 21]
```diff
@@ -1,6 +1,6 @@
 Rails.application.routes.draw do
-  root 'static_pages#home'
-
-  get 'help' => 'static_pages#help'
-
+  root             'static_pages#home'
+  get 'help'    => 'static_pages#help'
+  
+  resources :users
```
Comment (dieunb) [inline config/routes.rb]: use `only` or `except` to filter routes

**Pull request merged**